### PR TITLE
GBS Docs: Improving subgraph docs

### PR DESCRIPTION
**Pull request by trbromley**
Polishing and expanding docstrings for the `subgraph` module.

### CODE REVIEW

File: strawberryfields/gbs/subgraph.py [hunk 1]
```diff
@@ -19,25 +19,43 @@
 
 .. currentmodule:: strawberryfields.gbs.subgraph
 
-This module provides tools for users to identify dense subgraphs.
+This module provides tools for users to search for dense subgraphs of a range of different sizes.
```
Comment (ixfoduap) [inline strawberryfields/gbs/subgraph.py]:
```suggestion
This module provides tools for users to search dense subgraphs of a range of different sizes.
```

File: strawberryfields/gbs/subgraph.py [hunk 2]
```diff
@@ -19,25 +19,43 @@
...
+The search heuristic in this module works by resizing a collection of starting subgraphs and
+keeping track of the densest identified. The starting subgraphs can be selected by sampling from
+GBS, resulting in candidates that are likely to be dense :cite:`arrazola2018using`.
 
-The :func:`search` function provides a heuristic algorithm for finding dense regions and proceeds
-by greedily resizing input subgraphs and keeping track of the densest found.
+An accompanying tutorial can be found :ref:`here <gbs-subgraph-tutorial>`.
+
+Heuristic
```
Comment (ixfoduap) [inline strawberryfields/gbs/subgraph.py]:
```suggestion
Algorithm
```

File: strawberryfields/gbs/subgraph.py [hunk 3]
```diff
@@ -19,25 +19,43 @@
...
+---------
+
+The heuristic algorithm provided proceeds as follows. Each starting subgraph :math:`s` is resized
+to a range of sizes :math:`\{k\}_{k_{\min}}^{k_{\max}}`, resulting in the resized subgraphs
```
Comment (ixfoduap) [inline strawberryfields/gbs/subgraph.py]:
```suggestion
to a range of sizes :math:`\{k\}_{k_{\min}}^{k_{\max}}`, resulting in the subgraphs
```
Too many "sizes" and "resizes" :)

File: strawberryfields/gbs/subgraph.py [hunk 4]
```diff
@@ -177,14 +210,26 @@ def resize(subgraph: list, graph: nx.Graph, min_size: int, max_size: int) -> dic
 
     When growth is required, the algorithm examines all nodes from the remainder of the graph as
     candidates and adds-in the single node with the highest degree relative to the rest of the
```
Comment (ixfoduap) [inline strawberryfields/gbs/subgraph.py]:
```suggestion
    candidates and adds the single node with the highest degree relative to the rest of the
```

File: strawberryfields/gbs/subgraph.py [hunk 5]
```diff
@@ -177,14 +210,26 @@ def resize(subgraph: list, graph: nx.Graph, min_size: int, max_size: int) -> dic
...
-    subgraph. This results in a graph that is one node larger,  and if growth is still required
-    the algorithm performs the above procedure again.
+    subgraph. This results in a graph that is one node larger, and if growth is still required
```
Comment (ixfoduap) [inline strawberryfields/gbs/subgraph.py]:
```suggestion
    subgraph. This results in a graph that is one node larger, and if growth is still required,
```

File: strawberryfields/gbs/subgraph.py (hunk 1 above)
Comment (trbromley) [reply strawberryfields/gbs/subgraph.py]: Is getting rid of the "for" correct here? "search dense subgraphs" (i.e., without the for) sounds like looking within the dense subgraphs themselves.

Comment (ixfoduap) [reply strawberryfields/gbs/subgraph.py]: How about changing "search" to "find"?

Comment (trbromley) [reply strawberryfields/gbs/subgraph.py]: Great!

**Pull request merged**